Arrowgene.Ddon.GameServer/Handler/InnStayInnHandler.cs

```diff
@@ -24,15 +20,12 @@ public override void Handle(GameClient client, StructurePacket<C2SInnStayInnReq>
             WalletType priceWalletType = InnGetStayPriceHandler.PointType;
             uint price = InnGetStayPriceHandler.Point;
 
-            // Update character wallet
-            CDataWalletPoint wallet = client.Character.WalletPointList.Where(wp => wp.Type == priceWalletType).Single();
-            wallet.Value = (uint) Math.Max(0, (int)wallet.Value - (int)price);
-            Database.UpdateWalletPoint(client.Character.CharacterId, wallet);
+            var walletUpdate = Server.WalletManager.RemoveFromWallet(client.Character, priceWalletType, price);
 
             client.Send(new S2CInnStayInnRes()
             {
-                PointType = wallet.Type,
-                Point = wallet.Value
+                PointType = priceWalletType,
+                Point = walletUpdate?.Value ?? 0
```

Comment:
Why null coalesce? Should we really fail silently if this were to happen?

Reply:
Yes, but thats clearly an error situation right? that shouldnt ever happen, and if it does we might want to handle it in some way (return an error packet, log it, whatever)